### PR TITLE
speed up uid generation

### PR DESCRIPTION
**Pull request by andre-merzky**
test: generating 1m simple IDs
before:
```shell
(ve3)  rivendell  merzky  ~/radical/radical.utils.devel  [optimize/generate_id *] $ time ./t.py
16.283207654953003
r:0m16.415s  u:0m14.472s  s:0m1.941s
```

after:
```shell
(ve3)  rivendell  merzky  ~/radical/radical.utils.devel  [optimize/generate_id] $ time ./t.py
3.7440223693847656
r:0m3.909s  u:0m3.877s  s:0m0.032s
```

runtime reduced to ~25%, but more importantly system load reduced to ~2% due to reduced number of hits on the file system.  This has a much larger impact on shared file systems.

### CODE REVIEW

File: src/radical/utils/ids.py [hunk 1]
```diff
@@ -272,10 +283,6 @@ def _generate_id(template, prefix, ns=None):
       # print ret
         raise ValueError('unknown pattern in template (%s)' % template)
 
-    if 'client_notify' in ret and 'get' in ret:
```
Comment (mturilli) [inline src/radical/utils/ids.py]: We don't need this anymore?

File: src/radical/utils/ids.py [hunk 2]
```diff
@@ -272,10 +283,6 @@ def _generate_id(template, prefix, ns=None):
       # print ret
```
Comment (mturilli) [inline src/radical/utils/ids.py]: Irrelevant: alignment?

File: src/radical/utils/ids.py (hunk 1 above)
Comment (andre-merzky) [reply src/radical/utils/ids.py]: Thanks for checking!  No, that was debug code which already should have been removed (was never triggered though).

File: src/radical/utils/ids.py (hunk 2 above)
Comment (andre-merzky) [reply src/radical/utils/ids.py]: thanks, fixed.

**Pull request merged**